File: BagliListe1.cpp
#include<iostream>
using namespace std;

// tek yonlu bagli liste isimizi gorecektir.

class Dugum
{
public: 
	int data;
	Dugum* next;

	Dugum()
	{
		data = NULL;
		next = NULL;
	}
};


void Ekle(Dugum*, int);

void Bas(Dugum*);

Dugum* kok = NULL;

int main()
{
		 
	int index = NULL;

	cout << "Kac eleman gireceksiniz :";
	
	cin >> index;
	
	system("cls");
	
	cout << "Elemanlar :";

	int* dizi = new int[index];		// heap alanda girilen index kadar dizi olustur.

	for (int i = 0;i < index;i++)cin >> dizi[i];		// kullanicidan elemanlari al

	system("cls");

	for (int i = 0;i < index;i++)
	{
		Ekle(kok, dizi[i]);		// her bir elemani bagli listeye ekle
	}

	Bas(kok);		// diziyi ekrana bas

	delete[] dizi;		// heap alandaki diziyi kaldir
	delete kok;			// bagli listeyi heap den kaldir.


	return 0;
}
void Ekle(Dugum* lb, int veri)
{
	Dugum* yeni = new Dugum();

	yeni->data = veri;
	
	Dugum* iter = lb;

	if (kok == NULL)
	{
		kok = yeni;
		return;
	}

	while (iter->next != NULL)
	
	{
	
		iter = iter->next;
	}

	iter->next = yeni;
	

}
void Bas(Dugum* lb)
{
	Dugum* iter = lb;
	
	while (iter!= NULL)
	{
		cout << iter->data << " ";
	
		iter = iter->next;
	} 
	cout << endl;
}

File: BagliListe1.py
class Node:
    def __init__(self, data="", nextt=""):
        self.data = data
        self.nextt = nextt

def bastir(r):
    while r != None:
        print(r.data, end=" ")
        r = r.nextt

def ekle(r, v):
    while r.nextt != None:
        r = r.nextt
    
    r.nextt = Node()
    r.nextt.data = v
    r.nextt.nextt = None

dizi = [2, 3, 6, 3, 4, 1, 9, 8, 2, 6]

root = Node()
root.data = dizi[0]
root.nextt = None

for i in range(1, len(dizi)):
    ekle(root, dizi[i])

bastir(root)

# OUTPUT:
# 2 3 6 3 4 1 9 8 2 6 
